### PR TITLE
Repop and lost-unfound overhaul

### PR DESCRIPTION
**Pull request by athanatos**
mark_lost_revert/delete can race dangerously with normal IO due to not being ordered with other repops.  This series generalizes the repop ordering machinery and introduces a new message for performing mark_unfound_revert/delete.

### CODE REVIEW

**Comment by athanatos** [comment]
qa-suite PR: https://github.com/ceph/ceph-qa-suite/pull/832

Test run: samuelj-2016-02-22_06:57:34-rados-wip-sam-working-distro-basic-smithi

Ready for merge.

File: src/osd/ReplicatedPG.h [hunk 1]
```diff
@@ -893,12 +876,33 @@ class ReplicatedPG : public PG, public PGBackend::Listener {
...
+    ceph_tid_t rep_tid);
+  RepGather *new_repop(
+    ObcLockManager &&manager,
+    boost::optional<std::function<void(void)> > &&on_complete);
   void remove_repop(RepGather *repop);
 
-  RepGather *simple_repop_create(ObjectContextRef obc);
-  void simple_repop_submit(RepGather *repop);
+  OpContextUPtr simple_opc_create(ObjectContextRef obc);
+  void simple_opc_submit(OpContextUPtr ctx);
```
Comment (liewegas) [inline src/osd/ReplicatedPG.h]: all of the other methods use 'ctx' to mean OpContext (e.g., finish_ctx).. any specific reason to use opc here?

Comment (athanatos) [inline src/osd/ReplicatedPG.h]: These two are sort of external interfaces to the other *ctx* methods.  I could switch it if you feel strongly.

File: src/os/ObjectStore.h [hunk 2]
```diff
@@ -1415,6 +1417,23 @@ class ObjectStore {
       }
       data.ops++;
     }
+    void try_rename(coll_t cid, const ghobject_t& oldoid,
+                    const ghobject_t& oid) {
+      if (use_tbl) {
```
Comment (yuyuyu101) [inline src/os/ObjectStore.h]: do we already have a barrier feature prevent "use_tbl" is true?

**Comment by athanatos** [comment]
Passed tests after rebase:
sjust@teuthology:/a/samuelj-2016-02-27_00:36:36-rados-wip-sam-working-distro-basic-smithi

@liewegas Ok to merge?

**Comment by liewegas** [comment]
yep!

File: src/osd/PG.cc [hunk 3]
```diff
@@ -4554,39 +4556,71 @@ void PG::share_pg_info()
...
+      info.last_backfill_bitwise,
+      entries,
+      NULL,
+      pmissing,
+      NULL,
+      this);
+    pinfo.last_update = info.last_update;
+    pinfo.stats.stats_invalid = true;
+  }
+  for (auto &&i: entries) {
```
Comment (tchaikov) [inline src/osd/PG.cc]: `&&` does not make `i` a rvalue reference.

**Pull request merged**